alpha/apps/kaltura/lib/authenticationUtils.php

```diff
@@ -118,4 +95,28 @@ public static function verify2FACode($loginData, $otp)
 		$userSeed = $loginData->getSeedFor2FactorAuth();
 		return GoogleAuthenticator::verifyCode ($userSeed, $otp);
 	}
+	
+	protected static function addDynamicContentAuthMailJob($partnerId, $kuser, $loginData, $roleNames, $roleNameToUseDynamicEmailTemplate, $mailType)
+	{
+		$dynamicQrPageBaseLink = kEmails::getDynamicTemplateBaseLink($roleNames, kEmails::DYNAMIC_EMAIL_2FA_BASE_LINK);
```

Comment:
why did you pass roleName and not roleNameToUseDynamicEmailTemplate
akso would just call it userRole inside the function